src/qibo/transpiler/router.py

```diff
@@ -875,6 +888,39 @@ def _execute_blocks(self, blocklist: list):
         self._update_front_layer()
         self._memory_map = []
         self._delta_register = [1.0 for _ in self._delta_register]
+        self._added_swaps = 0
+        self._saved_circuit = deepcopy(self.circuit)
+
+    def _route_to_nearest_gate(self):
+        """Route the circuit to the nearest gate by adding SWAPs.
+
+        Method works in-place.
+        """
+
+        min_distance = float("inf")
+        shortest_path = None
+
+        for block in self._front_layer:
+            Q1 = self.circuit.get_physical_qubits(block)[0]
+            Q2 = self.circuit.get_physical_qubits(block)[1]
+            path = nx.bidirectional_shortest_path(self.connectivity, Q1, Q2)
+
+            # Between the gates in the front layer, the one requiring the minimum #SWAPs is selected
+            if len(path) < min_distance:
```

Comment:
Maybe we can first use a cycle to find the shortest path using "self._dist_matrix" and then compute the actual shortest path. In this way we call "nx.bidirectional_shortest_path()" only once.